@kiva/kv-shop/src/basket.ts

```diff
@@ -10,6 +13,34 @@ export function getBasketID() {
 	return getCookieValue('kvbskt');
 }
 
-export function setBasketID(basketId: string) {
+export function setBasketID() {
 	// TODO
 }
+
+export async function createBasket(apollo) {
+	try {
+		apollo.mutate({
+			mutation: gql`mutation createNewBasketForUser { shop { id createBasket } }`,
+		}).then(({ data }) => {
+			const newBasketId = data.shop?.createBasket ?? null;
+			if (newBasketId) {
+				document.cookie = `kvbskt=${newBasketId}; path=/; secure=true;`;
+			}
+		});
+	} catch (error) {
+		throw parseShopError(error);
+	}
+}
+
+export function hasBasketExpired(errorCode) {
+	return ['shop.invalidBasketId', 'shop.basketRequired'].includes(errorCode);
+}
+
+export async function handleInvalidBasketForDonation({ donationAmount, navigateToCheckout = false, apollo }) {
```

Comment:
While it won't be trivial, long-term we'll want unit tests on these new methods. Please at least add a ticket to come back and address.

Reply:
Same with the `basketItems` file. Looks like that file has no tests.

Reply:
Added some basic tests but left a comment to add remaining cases